code/modules/lighting/light_effect.dm

```diff
@@ -0,0 +1,118 @@
+/atom/movable/light
+	mouse_opacity = 0
+	plane = LIGHTING_PLANE_MASTER
+
+	layer = 1
+	//layer 1 = base plane layer
+	//layer 2 = base shadow templates
+	//layer 3 = wall lighting overlays
+	//layer 4 = light falloff overlay
+
+	appearance_flags = KEEP_TOGETHER
+	icon = null
+	invisibility = INVISIBILITY_LIGHTING
+	pixel_x = -WORLD_ICON_SIZE*2
+	pixel_y = -WORLD_ICON_SIZE*2
+	glide_size = WORLD_ICON_SIZE
+	blend_mode = BLEND_ADD
+
+	alpha = 180
+
+	var/current_power = 1
+	var/atom/movable/holder
+	var/point_angle
+	var/list/affecting_turfs = list()
+	var/list/temp_appearance
+
+/atom/movable/light/shadow
+	appearance_flags = KEEP_TOGETHER | TILE_BOUND
+
+/atom/movable/light/New(var/newholder)
```

Comment:
I don't really like changing the behavior of the first arg to `New()`, but do it if you must, I guess.